lib/azure/armrest/resource_service.rb

```diff
@@ -23,16 +26,29 @@ def initialize(configuration, options = {})
       def list(resource_group, options = {})
         url = build_url(resource_group, options)
         response = rest_get(url)
-        Azure::Armrest::ArmrestCollection.create_from_response(response, Azure::Armrest::Resource)
+
+        if options[:all]
+          get_all_results(response)
+        else
+          Azure::Armrest::ArmrestCollection.create_from_response(response, Azure::Armrest::Resource)
+        end
       end
 
       # Same as Azure::Armrest::ResourceService#list but returns all resources
       # for all resource groups.
       #
+      # Normally this is capped at 1000 results, but you may specify the :all
+      # option to get everything.
+      #
       def list_all(options = {})
         url = build_url(nil, options)
         response = rest_get(url)
-        Azure::Armrest::ArmrestCollection.create_from_response(response, Azure::Armrest::Resource)
+
+        if options[:all]
+          get_all_results(response)
```

Comment:
Well, the other resource services only provide a subset of information, so I'm not generally worried about huge datasets being returned. Since this returns information for *all* resources, I felt it was warranted here.